ocaml/xenopsd/xc/device.ml

```diff
@@ -2091,10 +2091,11 @@ module Dm_Common = struct
   let cmdline_of_disp ?domid info =
     let vga_type_opts x =
       let open Xenops_interface.Vgpu in
+      (* We can match on the implementation details to detect the VCS
+         case. Don't pass -vgpu for a compute vGPU. *)
       match x with
-      | Vgpu [{implementation= Nvidia _; _}] ->
-          ["-vgpu"]
-      | Vgpu ({implementation= Nvidia _; _} :: _) ->
+      | Vgpu ({implementation= Nvidia {vclass; _}; _} :: _)
+        when vclass <> Some "Compute" ->
```

Comment:
This change will make compute vgpus to raise an exception. It needs another case added for them to return an empty list, probably along the MxGPU case

Reply:
What exception? No match?

Reply:
Will rewrite this